File: EntreBicis_Backend/src/main/java/cat/copernic/mbotana/entrebicis_backend/entity/Route.java
package cat.copernic.mbotana.entrebicis_backend.entity;

import java.time.LocalTime;
import java.util.List;

import cat.copernic.mbotana.entrebicis_backend.entity.enums.RouteState;
import cat.copernic.mbotana.entrebicis_backend.entity.enums.RouteValidate;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Route {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private RouteState routeState;

    @Column
    private RouteValidate isValidate;

    @Column
    private Double totalRutePoints;

    @Column
    private Double totalRouteDistance;

    @Column 
    private LocalTime totalRouteTime;

    @Column
    private Double maxRouteVelocity;

    @Column 
    private Double avgRouteVelocity;

    @OneToMany(mappedBy = "route")
    private List<GpsPoint> gpsPoints;

    @ManyToOne
    @JoinColumn(name = "user_email", unique = true)
    private User user;

}

File: EntreBicis_Frontend/app/src/main/java/cat/copernic/mbotana/entrebicis_frontend/route_management/domain/models/Route.kt
package cat.copernic.mbotana.entrebicis_frontend.route_management.domain.models

import cat.copernic.mbotana.entrebicis_frontend.core.enums.RouteState
import cat.copernic.mbotana.entrebicis_frontend.core.enums.RouteValidate
import cat.copernic.mbotana.entrebicis_frontend.gpsPoint_management.domain.models.GpsPoint
import cat.copernic.mbotana.entrebicis_frontend.user_management.domain.models.User
import java.time.LocalTime

data class Route(
    var id: Long? = null,
    var routeState: RouteState,
    var routeValidate: RouteValidate,
    var totalRoutePoints: Double,
    var totalRouteDistance: Double,
    var totalRouteTime: LocalTime,
    var maxRouteVelocity: Double,
    var avgRouteVelocity: Double,
    var gpsPoints: List<GpsPoint>,
    var user: User
)
